crates/hir-def/src/data.rs

```diff
@@ -567,6 +571,8 @@ pub struct StaticData {
     pub visibility: RawVisibility,
     pub mutable: bool,
     pub is_extern: bool,
+    pub has_safe_kw: bool,
+    pub has_unsafe_kw: bool,
```

Comment:
I don't think we promise pretty-printing will recover exactly the original code, at least if we do there are bunch of cases this promise is broken.

Reply:
And I'd argue it's better to always write `unsafe` in pretty printing since it may not be clear that the function is unsafe because it is extern (if we even show that it is extern, I don't remember now).

Reply:
I kept both `has_xxx_kw` here in case we need them later, and it is similar to `FnFlags`. I also left a comment here: https://github.com/roife/rust-analyzer/blob/002f6ad6f1ca6aa59e1ebccc8478930101f8150f/crates/hir-def/src/item_tree.rs#L826-L829 that we may use `bitflags` in the future.

Reply:
This one shouldn't have changed the size, but the next one (5th bool) might given alignment of 8 and other 4 byte fields in this struct.

I do think we should just swap all of these kinds of flags (this struct and others) to use bitflags, its more uniform that way and generally packs nicer.

Reply:
I do agree we don't really need to preserve the explicit `unsafe` in our IRs here though, as in this case its not meaningful. But it doesn't really matter much either